File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnContributorEdit.cs

namespace GracenoteSDK {

/**
*  @internal GnContributorEdit @endinternal
*  represents  GnContributor's information returned by Gracenote.
*/
public class GnContributorEdit : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnContributorEdit(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnContributorEdit_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnContributorEdit obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnContributorEdit() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnContributorEdit(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public void Genre(GnListElement genreElement) {
    gnsdk_csharp_marshalPINVOKE.GnContributorEdit_Genre(swigCPtr, GnListElement.getCPtr(genreElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Origin(GnListElement originElement) {
    gnsdk_csharp_marshalPINVOKE.GnContributorEdit_Origin(swigCPtr, GnListElement.getCPtr(originElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Era(GnListElement eraElement) {
    gnsdk_csharp_marshalPINVOKE.GnContributorEdit_Era(swigCPtr, GnListElement.getCPtr(eraElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void ArtistType(GnListElement arttypeElement) {
    gnsdk_csharp_marshalPINVOKE.GnContributorEdit_ArtistType(swigCPtr, GnListElement.getCPtr(arttypeElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public GnContributor GnContributor {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnContributorEdit_GnContributor_get(swigCPtr);
      GnContributor ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnContributor(cPtr, true);
      return ret;
    } 
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnContributorEdit.java

package com.gracenote.gnsdk;

/** 
* 
*  represents  GnContributor's information returned by Gracenote. 
*/ 
 
public class GnContributorEdit extends GnDataObject {
  private transient long swigCPtr;

  protected GnContributorEdit(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnContributorEdit_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnContributorEdit obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnContributorEdit(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnContributor gnContributor() {
    return new GnContributor(gnsdk_javaJNI.GnContributorEdit_gnContributor(swigCPtr, this), true);
  }

  public void genre(GnListElement genreElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnContributorEdit_genre(swigCPtr, this, GnListElement.getCPtr(genreElement), genreElement);
  }

  public void origin(GnListElement originElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnContributorEdit_origin(swigCPtr, this, GnListElement.getCPtr(originElement), originElement);
  }

  public void era(GnListElement eraElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnContributorEdit_era(swigCPtr, this, GnListElement.getCPtr(eraElement), eraElement);
  }

  public void artistType(GnListElement arttypeElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnContributorEdit_artistType(swigCPtr, this, GnListElement.getCPtr(arttypeElement), arttypeElement);
  }

}
